tool/tsh/db.go

```diff
@@ -287,7 +287,7 @@ func onDatabaseConnect(cf *CLIConf) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	profile, err := client.StatusCurrent("", cf.Proxy)
+	profile, err := client.StatusCurrentOfCluster("", cf.Proxy, cf.SiteName)
```

Comment:
I looks that extension of `client.StatusCurrent`  function by `cf.SiteName` param  is superfluous. When you will track the `profile` var returned by `client.StatusCurrentOfCluster("", cf.Proxy, cf.SiteName)` function in 
in app/db handlers like: `onListDatabases`, `onDatabaseConnect`, `databaseLogin`, `onAppLogin`, `onAppConfig` you will notice that there is no any reference to `profile.SiteName` in these handlers.  

The only places where cf.SiteName is actually used in when the teleport client is created https://github.com/gravitational/teleport/blob/master/tool/tsh/tsh.go#L1811
and when app/login command issuing the user cert to proper clusters: 
https://github.com/gravitational/teleport/blob/7606d330e9aed600f02607a3c723c31d18368a50/tool/tsh/app.go#L72
https://github.com/gravitational/teleport/blob/cc3c38d7805e007b25afe751d3690f4e51ba0a0f/tool/tsh/db.go#L106

Reply:
`StatusCurrent` returns/generates the path to the certs, with the cluster name saved in the profile yaml.

Without the cluster name overwrite, I have trouble that it gives wrong path to pems. Or it won't find existing apps/dbs certs. 

What would you suggest?

Reply:
> Without the cluster name overwrite, I have trouble that it gives wrong path to pems. Or it won't find existing apps/dbs certs.
 
 Right I have missed this place though in current approach we are trying to load whole cluster user key in `readProfile` call thus if user is not logged into leaf cluster the --cluster flag in db/app commands won't work. 

For instance: 
```
rm -rf ~/.tsh
# Login into root cluster
tsh login --proxy=ice-berg.dev:3080 --user=marek

# Try to list first-leaf cluster databases
tsh db --cluster=first-leaf ls
ERROR: not logged in to cluster first-leaf

# try to connect to instance1 db on first-leaf  cluster 
tsh db --cluster=first-leaf  connect instance1
ERROR: not logged in to cluster first-leaf
```

where ssh access to leaf cluster works properly: 
```
m -rf ~/.tsh
# Login into root cluster
tsh login --proxy=ice-berg.dev:3080 --user=marek

# List ssh leaf cluster nodes
tsh ls --cluster=first-leaf
Node Name       Address         Labels
--------------- --------------- -------
node-first-leaf 127.0.0.1:33022 env=dev

# SSH to node in leaf cluster
tsh  ssh --cluster=first-leaf  node-first-leaf
```

So right now without logging into leaf cluster the app/db `--cluster` flag doesn't work properly.

Reply:
yes, my change tries to workaround the current/old `readProfile`.

if leaf cluster is not logged in yet with `tsh login <leaf>`, `app/db --cluster` will prompt `not logged in to cluster <leaf>`

if leaf cluster is logged in, `readProfile` will get the cluster name overwrite and load keys for what's specified in `--cluster`

Reply:
> if leaf cluster is not logged in yet with tsh login <leaf>, app/db --cluster will prompt not logged in to cluster <leaf>

All right, lets confirm with @r0mant that that is expected behaviour.

Reply:
@greedy52 @smallinsky I think we should make this consistent with how other commands that take `--cluster` flag behave. For example, I think `tsh --cluster=leaf ls` command doesn't require you to log into leaf first - if so, we should keep the same behavior.